From 4da56227f47a96f37eb0e46e0222f7e1d2f95c2b Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:32:16 -0400
Subject: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/openssh.spec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SPECS/openssh.spec b/SPECS/openssh.spec
index 0f39971..d92504f 100644
--- a/SPECS/openssh.spec
+++ b/SPECS/openssh.spec
@@ -73,7 +73,7 @@
 Summary: An open source implementation of SSH protocol version 2
 Name: openssh
 Version: %{openssh_ver}
-Release: %{openssh_rel}%{?dist}%{?rescue_rel}
+Release: %{openssh_rel}.el8%{?_tis_dist}.%{tis_patch_ver}
 URL: http://www.openssh.com/portable.html
 #URL1: http://pamsshagentauth.sourceforge.net
 Source0: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz
@@ -323,7 +323,7 @@ Requires: openssh = %{version}-%{release}
 Summary: PAM module for authentication with ssh-agent
 Group: System Environment/Base
 Version: %{pam_ssh_agent_ver}
-Release: %{pam_ssh_agent_rel}.%{openssh_rel}%{?dist}%{?rescue_rel}
+Release: %{pam_ssh_agent_rel}.el8%{?_tis_dist}.%{tis_patch_ver}
 License: BSD
 
 %description
-- 
1.8.3.1

