From 5276ac3b8b6e8a90de670290c03978640fceb20f Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Sat, 2 Nov 2019 17:15:03 +0800
Subject: [PATCH 1/2] Update-package-versioning-for-TIS-format

---
 SPECS/openssh.spec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SPECS/openssh.spec b/SPECS/openssh.spec
index 0f39971..d92504f 100644
--- a/SPECS/openssh.spec
+++ b/SPECS/openssh.spec
@@ -73,7 +73,7 @@
 Summary: An open source implementation of SSH protocol version 2
 Name: openssh
 Version: %{openssh_ver}
-Release: %{openssh_rel}%{?dist}%{?rescue_rel}
+Release: %{openssh_rel}.el8%{?_tis_dist}.%{tis_patch_ver}
 URL: http://www.openssh.com/portable.html
 #URL1: http://pamsshagentauth.sourceforge.net
 Source0: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz
@@ -323,7 +323,7 @@ Requires: openssh = %{version}-%{release}
 Summary: PAM module for authentication with ssh-agent
 Group: System Environment/Base
 Version: %{pam_ssh_agent_ver}
-Release: %{pam_ssh_agent_rel}.%{openssh_rel}%{?dist}%{?rescue_rel}
+Release: %{pam_ssh_agent_rel}.el8%{?_tis_dist}.%{tis_patch_ver}
 License: BSD
 
 %description
-- 
1.8.3.1

