From 788beea32cfa09f61c27db117ec3b3b0500fbad8 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:32:16 -0400
Subject: spec-include-TiS-changes.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/openssh.spec | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/SPECS/openssh.spec b/SPECS/openssh.spec
index d92504f..cf85d8a 100644
--- a/SPECS/openssh.spec
+++ b/SPECS/openssh.spec
@@ -659,9 +659,6 @@ getent passwd sshd >/dev/null || \
 %preun server
 %systemd_preun sshd.service sshd.socket
 
-%postun server
-%systemd_postun_with_restart sshd.service
-
 %files
 %license LICENCE
 %doc CREDITS ChangeLog INSTALL OVERVIEW PROTOCOL* README README.platform README.privsep README.tun README.dns TODO
-- 
1.8.3.1

