From 85cda50f8efa9f7254aba38ebf1071dc5ecc851d Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Tue, 5 Nov 2019 17:32:42 +0800
Subject: [PATCH 2/5] Add-TIS-patches

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/anaconda.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index b345630..c4db2fd 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -15,6 +15,10 @@ URL:     http://fedoraproject.org/wiki/Anaconda
 Patch100: centos-l10n.patch
 Patch101: 0001-branding-syspurpose-doesn-t-make-sense-for-CentOS.patch
 Patch102: 0002-add-centos-background-color.patch
+
+# WRS
+Patch10001: 0001-TIS-Progress-and-error-handling.patch
+
 # To generate Source0 do:
 # git clone https://github.com/rhinstaller/anaconda
 # git checkout -b archive-branch anaconda-%%{version}-%%{release}
-- 
2.7.4

