From 103302403b10e2a97827e82a550466d3bc87015a Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Tue, 5 Nov 2019 17:34:45 +0800
Subject: [PATCH 3/5] revert-7.4-grub2-efi-handling

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/anaconda.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index c4db2fd..84fafb9 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -18,6 +18,7 @@ Patch102: 0002-add-centos-background-color.patch
 
 # WRS
 Patch10001: 0001-TIS-Progress-and-error-handling.patch
+Patch10002: 0002-revert-7.4-grub2-efi-handling.patch
 
 # To generate Source0 do:
 # git clone https://github.com/rhinstaller/anaconda
-- 
2.7.4

