From dcba6cc2b14c16ea6e079c536d371a6c522a0a7b Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Tue, 5 Nov 2019 17:35:44 +0800
Subject: [PATCH 4/5] Upversion-rpm-devel-dependency

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/anaconda.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index 84fafb9..a1d4854 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -47,7 +47,7 @@ Source0: %{name}-%{version}.tar.bz2
 %define nmver 1.0
 %define pykickstartver 3.16.2-1
 %define pypartedver 2.5-2
-%define rpmver 4.10.0
+%define rpmver 4.14.0
 %define simplelinever 1.1-1
 %define utillinuxver 2.15.1
 
-- 
2.7.4

