From 2991d9310c342a937ecdb4c3a4dd11751301bf2d Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Tue, 5 Nov 2019 17:41:03 +0800
Subject: [PATCH 5/5] Add-TIS-patches-for-host-lookup

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/anaconda.spec | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index a1d4854..a379ded 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -19,6 +19,8 @@ Patch102: 0002-add-centos-background-color.patch
 # WRS
 Patch10001: 0001-TIS-Progress-and-error-handling.patch
 Patch10002: 0002-revert-7.4-grub2-efi-handling.patch
+Patch10003: 0003-Set-default-hostname-to-localhost.patch
+Patch10004: 0004-Cache-server-ip-in-etc-hosts.patch
 
 # To generate Source0 do:
 # git clone https://github.com/rhinstaller/anaconda
@@ -267,6 +269,10 @@ mkdir %{buildroot}%{_datadir}/anaconda/addons
 %ifarch %livearches
 desktop-file-install --dir=%{buildroot}%{_datadir}/applications %{buildroot}%{_datadir}/applications/liveinst.desktop
 %endif
+
+# Add anaconda-preexec script
+install -m 0755 scripts/anaconda-preexec %{buildroot}%{_sbindir}/anaconda-preexec
+
 # NOTE: If you see "error: Installed (but unpackaged) file(s) found" that include liveinst files,
 #       check the IS_LIVEINST_ARCH in configure.ac to make sure your architecture is properly defined
 
@@ -324,6 +330,7 @@ update-desktop-database &> /dev/null || :
 %{_datadir}/applications/*.desktop
 %{_sysconfdir}/xdg/autostart/*.desktop
 %endif
+%{_sbindir}/anaconda-preexec
 
 %files gui
 %{python3_sitearch}/pyanaconda/ui/gui/*
-- 
2.7.4

