From 0330ca87e6b99d145d054cb88be6e51c156c762b Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Thu, 31 Oct 2019 18:15:25 +0800
Subject: [PATCH 1/2] Update-package-versioning-for-TIS-format

---
 SPECS/tpm2-tools.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/tpm2-tools.spec b/SPECS/tpm2-tools.spec
index 8673838..c0a9d64 100644
--- a/SPECS/tpm2-tools.spec
+++ b/SPECS/tpm2-tools.spec
@@ -1,6 +1,6 @@
 Name: tpm2-tools
 Version: 3.1.1
-Release: 4%{?dist}
+Release: 4.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary: A TPM2.0 testing tool build upon TPM2.0-TSS
 
 License: BSD
-- 
1.8.3.1

