From 1f2ee4452b6d966e28dcdb92db21451d2581fb11 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Thu, 31 Oct 2019 18:16:35 +0800
Subject: [PATCH 2/2] disable-socket-mode-TCTI

---
 SPECS/tpm2-tools.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/tpm2-tools.spec b/SPECS/tpm2-tools.spec
index c0a9d64..f600d3a 100644
--- a/SPECS/tpm2-tools.spec
+++ b/SPECS/tpm2-tools.spec
@@ -35,7 +35,7 @@ tpm2-tools is a batch of testing tools for tpm2.0. It is based on tpm2-tss.
 %autosetup -p1 -n %{name}-%{version}
 
 %build
-%configure --prefix=/usr --disable-static --disable-silent-rules
+%configure --prefix=/usr --disable-static --disable-silent-rules --without-tcti-socket
 %make_build
 
 %install
-- 
1.8.3.1

