From 27b9c7a5281c84da9f9029deeb31442cf17f5755 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:42:22 -0400
Subject: [PATCH 2/2] WRS: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/net-tools.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/net-tools.spec b/SPECS/net-tools.spec
index 4e5041b..82dfa5e 100644
--- a/SPECS/net-tools.spec
+++ b/SPECS/net-tools.spec
@@ -3,7 +3,7 @@
 Summary: Basic networking tools
 Name: net-tools
 Version: 2.0
-Release: 0.51.%{checkout}%{?dist}
+Release: 0.51.20160912git.el8%{?_tis_dist}.%{tis_patch_ver}
 License: GPLv2+
 Group: System Environment/Base
 URL: http://sourceforge.net/projects/net-tools/
-- 
1.8.3.1

