From 24d92c7a7730e71bac4182df53d5fd6f4d7d6957 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:42:22 -0400
Subject: [PATCH 1/2] WRS: spec-to-include-TiS-patches.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/net-tools.spec | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/SPECS/net-tools.spec b/SPECS/net-tools.spec
index 82dfa5e..797e7e7 100644
--- a/SPECS/net-tools.spec
+++ b/SPECS/net-tools.spec
@@ -38,6 +38,10 @@ Patch22: net-tools-timer-man.patch
 Patch23: net-tools-interface-name-len.patch
 Patch24: net-tools-covscan.patch
 
+# WindRiver patches
+Patch100: net-tools-hostname-ipv6-shortname.patch
+Patch101: net-tools-ifconfig-no-ifstate-on-flush.patch
+
 BuildRequires: bluez-libs-devel
 BuildRequires: gettext, libselinux
 BuildRequires: libselinux-devel
@@ -70,6 +74,9 @@ cp %SOURCE8 ./man/en_US
 %patch23 -p1 -b .interface-name-len
 %patch24 -p1 -b .covscan
 
+%patch100 -p1 -b .hostname-ipv6
+%patch101 -p1 -b .ifconfig-no-ifstate-on-flush
+
 touch ./config.h
 
 %build
-- 
1.8.3.1

