From 512fcc9bc1d0aa84677d09234cd4243e027f37e7 Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Fri, 8 Nov 2019 14:57:14 +0800
Subject: [PATCH 2/2] dhclient-ipv6-bind-to-interface

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 common/socket.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/common/socket.c b/common/socket.c
index 94ce334..17d32e4 100644
--- a/common/socket.c
+++ b/common/socket.c
@@ -234,6 +234,15 @@ if_register_socket(struct interface_info *info, int family,
 	}
 #endif
 
+#if defined(SO_BINDTODEVICE)
+	/* Bind this socket to this interface. */
+	if ((!do_multicast || !*do_multicast) && info->ifp &&
+	    setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE,
+			(char *)(info -> ifp), sizeof(*(info -> ifp))) < 0) {
+		log_error("setsockopt: SO_BINDTODEVICE: %m");
+	}
+#endif
+
 	/* Bind the socket to this interface's IP address. */
 	if (bind(sock, (struct sockaddr *)&name, name_len) < 0) {
 		log_error("Can't bind to dhcp address: %m");
@@ -244,15 +253,6 @@ if_register_socket(struct interface_info *info, int family,
 		log_fatal("includes a bootp server.");
 	}
 
-#if defined(SO_BINDTODEVICE)
-	/* Bind this socket to this interface. */
-	if ((local_family != AF_INET6) && (info->ifp != NULL) &&
-	    setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE,
-			(char *)(info -> ifp), sizeof(*(info -> ifp))) < 0) {
-		log_fatal("setsockopt: SO_BINDTODEVICE: %m");
-	}
-#endif
-
 	/* IP_BROADCAST_IF instructs the kernel which interface to send
 	 * IP packets whose destination address is 255.255.255.255.  These
 	 * will be treated as subnet broadcasts on the interface identified
-- 
2.7.4

