From 9e05fbda393fe0ed6cb581c33ba19a14ab07abb6 Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Wed, 13 Dec 2017 09:02:08 -0500
Subject: [PATCH 1/2] Update package versioning for TIS format

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/python-requests.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python-requests.spec b/SPECS/python-requests.spec
index 3c980d3..7d1f842 100644
--- a/SPECS/python-requests.spec
+++ b/SPECS/python-requests.spec
@@ -10,7 +10,7 @@
 
 Name:           python-requests
 Version:        2.20.0
-Release:        1%{?dist}
+Release:        1.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary:        HTTP library, written in Python, for human beings
 
 License:        ASL 2.0
-- 
1.8.3.1

