From b99292ef32fdde60d2ec4181ee5963b93a97cdf2 Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Wed, 13 Dec 2017 09:02:40 -0500
Subject: [PATCH 2/2] include patch for error handling

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/python-requests.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/python-requests.spec b/SPECS/python-requests.spec
index 7d1f842..1c77f7c 100644
--- a/SPECS/python-requests.spec
+++ b/SPECS/python-requests.spec
@@ -38,6 +38,8 @@ Patch4:         Don-t-inject-pyopenssl-into-urllib3.patch
 #   build-time package tests
 Patch5:         Skip-all-tests-needing-httpbin.patch
 
+Patch6:         0001-close-connection-on-HTTP-413-Request-Entit.patch
+
 BuildArch:      noarch
 
 %description
-- 
1.8.3.1

