From 4ce897e0fb4638859086f72ef57abc4f3411eb9b Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Mon, 4 Nov 2019 11:18:11 +0800
Subject: [PATCH 1/9] grub2-Update-package-versioning-for-TIS-format

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/grub2.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index b471027..3e0bc91 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -7,7 +7,7 @@
 Name:		grub2
 Epoch:		1
 Version:	2.02
-Release:	66%{?dist}
+Release:	66.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary:	Bootloader with support for Linux, Multiboot and more
 Group:		System Environment/Base
 License:	GPLv3+
-- 
2.7.4

