From 432b359af25ebe96f1bfff2adab26f6a1b17c15c Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Mon, 4 Nov 2019 14:13:57 +0800
Subject: [PATCH 02/10] grub2-remove-debug-pkgs

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/grub2.spec | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 3e0bc91..c48b6fd 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -188,7 +188,6 @@ makeinfo --html --no-split -I docs -o docs/grub.html \
 %install
 set -e
 rm -fr $RPM_BUILD_ROOT
-
 %do_common_install
 %if 0%{with_efi_arch}
 %{expand:%do_efi_install %%{grubefiarch} %%{grubefiname} %%{grubeficdname}}
@@ -242,25 +241,6 @@ install -d -m 0755 %{buildroot}%{_unitdir}/system-update.target.wants
 ln -s ../grub-boot-indeterminate.service \
 	%{buildroot}%{_unitdir}/system-update.target.wants
 
-# Don't run debuginfo on all the grub modules and whatnot; it just
-# rejects them, complains, and slows down extraction.
-%global finddebugroot "%{_builddir}/%{?buildsubdir}/debug"
-
-%global dip RPM_BUILD_ROOT=%{finddebugroot} %{__debug_install_post}
-%define __debug_install_post (						\
-	mkdir -p %{finddebugroot}/usr					\
-	mv ${RPM_BUILD_ROOT}/usr/bin %{finddebugroot}/usr/bin		\
-	mv ${RPM_BUILD_ROOT}/usr/sbin %{finddebugroot}/usr/sbin		\
-	%{dip}								\
-	install -m 0755 -d %{buildroot}/usr/lib/ %{buildroot}/usr/src/	\
-	cp -al %{finddebugroot}/usr/lib/debug/				\\\
-		%{buildroot}/usr/lib/debug/				\
-	cp -al %{finddebugroot}/usr/src/debug/				\\\
-		%{buildroot}/usr/src/debug/ )				\
-	mv %{finddebugroot}/usr/bin %{buildroot}/usr/bin		\
-	mv %{finddebugroot}/usr/sbin %{buildroot}/usr/sbin		\
-	%{nil}
-
 %undefine buildsubdir
 
 %pre tools
-- 
2.7.4

