From c8a454eab9c3fbbe6fd165cc47159cd471f7496a Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Mon, 4 Nov 2019 14:58:08 +0800
Subject: [PATCH 6/9] grub2-Build-pxeboot-package

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SOURCES/grub.macros | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index 59d894d..ffc7113 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -278,6 +278,13 @@ Summary:    Unsigned versions of GRUB EFI binaries                      \
 %description %{1}-unsigned                                              \
 This package contains unsigned version of GRUB EFI binaries.            \
                                                                         \
+%package %{1}-pxeboot                                                   \
+Summary:    PXE bootable GRUB EFI binaries                              \
+                                                                        \
+%description %{1}-pxeboot                                               \
+This package contains the version of EFI GRUB that is served by the pxeboot \
+server                                                                  \
+                                                                        \
 %{nil}
 
 %global do_common_setup()					\
@@ -448,6 +455,7 @@ ln -sf ..%{efi_esp_dir}/grub.cfg				\\\
 	$RPM_BUILD_ROOT%{_sysconfdir}/%{name}-efi.cfg		\
 install -m 700 %{2} $RPM_BUILD_ROOT%{efi_esp_dir}/%{2}.unsigned \
 install -m 700 %{3} $RPM_BUILD_ROOT%{efi_esp_dir}/%{3}.unsigned \
+install -D -m 755 %{2} $RPM_BUILD_ROOT/pxeboot/EFI/%{2}         \
 install -m 700 %{2} $RPM_BUILD_ROOT%{efi_esp_dir}/%{2}		\
 install -m 700 %{3} $RPM_BUILD_ROOT%{efi_esp_dir}/%{3}		\
 install -D -m 700 unicode.pf2					\\\
@@ -554,4 +562,8 @@ touch ${RPM_BUILD_ROOT}/boot/%{name}/grub.cfg			\
 %{expand:%%files %{1}-unsigned}                                 \
 /boot/efi/EFI/%{efidir}/%{grubefiname}.unsigned                 \
 /boot/efi/EFI/%{efidir}/%{grubeficdname}.unsigned               \
+								\
+%{expand:%%files %{1}-pxeboot}                                  \
+/pxeboot/EFI/%{grubefiname}                                     \
+								\
 %{nil}
-- 
2.7.4

