From 2d3188b209d7796f738e8625f0c117e843ca244f Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Mon, 4 Nov 2019 15:09:39 +0800
Subject: [PATCH 9/9] grub2-Don-t-strip-img-files-from-non-EFI-build

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SOURCES/grub.macros | 2 --
 1 file changed, 2 deletions(-)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index 1255697..5f12320 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -528,8 +528,6 @@ touch ${RPM_BUILD_ROOT}/boot/%{name}/grub.cfg			\
 %defattr(-,root,root)						\
 %dir %{_libdir}/grub/%{2}/					\
 %{_libdir}/grub/%{2}/*						\
-%exclude %{_libdir}/grub/%{2}/*.module				\
-%exclude %{_libdir}/grub/%{2}/{boot,boot_hybrid,cdboot,diskboot,lzma_decompress,pxeboot}.image \
 %exclude %{_libdir}/grub/%{2}/*.o				\
 %else								\
 %%exclude %%{_libdir}/grub/%%{grublegacyarch}/*			\
-- 
2.7.4

