From 35e02f4d4cb0e874ef684a9bad47ee9ee3763292 Mon Sep 17 00:00:00 2001
From: chen-dq <chen.dq@neusoft.com>
Date: Fri, 15 Nov 2019 03:14:24 -0500
Subject: [PATCH 2/2] Update-spec-include-stx-patches

---
 SPECS/python-docker.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/python-docker.spec b/SPECS/python-docker.spec
index a2fa420..f9eb1dd 100644
--- a/SPECS/python-docker.spec
+++ b/SPECS/python-docker.spec
@@ -29,6 +29,7 @@ Source0:        https://files.pythonhosted.org/packages/source/d/%{srcname}/%{sr
 # work in RHEL.  This patch converts those environment markers into simple if
 # statements.
 Patch1: remove-environment-markers.patch
+Patch2: 0001-Update-client-close-socket.patch
 
 BuildArch:      noarch
 
@@ -138,7 +139,9 @@ run containers, manage containers, manage Swarms, etc.
 %endif # with_python3
 
 %prep
-%autosetup -n %{srcname}-%{version} -p 1
+%setup -q -n %{srcname}-%{version}
+%patch2 -p1
+
 rm -fr docker.egg-info
 
 %build
-- 
2.18.1

