From c25a30b4a0c7347234c2af4afab099b5735bbf71 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:05:59 -0400
Subject: [PATCH] WRS: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/net-snmp.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/net-snmp.spec b/SPECS/net-snmp.spec
index cd968e7..597e866 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -10,7 +10,7 @@
 Summary:    A collection of SNMP protocol tools and libraries
 Name:       net-snmp
 Version:    5.8
-Release:    7%{?dist}.1
+Release:    7.el8%{?_tis_dist}.%{tis_patch_ver}.1
 Epoch:      1
 
 License:    BSD
-- 
1.8.3.1

