From 369cdb6480a57a0adb9f39a8cc7acdefd9b00080 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 4 Nov 2019 10:25:55 +0800
Subject: [PATCH 3/3] openldap-enable-password-policy

---
 SPECS/openldap.spec | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/SPECS/openldap.spec b/SPECS/openldap.spec
index bca9358..ae51d1c 100644
--- a/SPECS/openldap.spec
+++ b/SPECS/openldap.spec
@@ -42,6 +42,9 @@ Patch24: openldap-openssl-manpage-defaultCA.patch
 Patch90: check-password-makefile.patch
 Patch91: check-password.patch
 
+# WRS specific patches
+Patch200: rootdn-should-not-bypass-ppolicy.patch
+
 BuildRequires: cyrus-sasl-devel, openssl-devel, krb5-devel, unixODBC-devel
 BuildRequires: glibc-devel, libtool, libtool-ltdl-devel, groff, perl-interpreter, perl-devel, perl-generators, perl(ExtUtils::Embed)
 
@@ -116,6 +119,8 @@ AUTOMAKE=%{_bindir}/true autoreconf -fi
 %patch22 -p1
 %patch23 -p1
 %patch24 -p1
+# WRS
+%patch200 -p1
 
 # build smbk5pwd with other overlays
 ln -s ../../../contrib/slapd-modules/smbk5pwd/smbk5pwd.c servers/slapd/overlays
@@ -209,12 +214,12 @@ install -d -m 740 %{buildroot}%{_sysconfdir}/openldap
 cat > %{buildroot}%{_sysconfdir}/openldap/check_password.conf <<EOF
 # OpenLDAP pwdChecker library configuration
 
-#useCracklib 1
-#minPoints 3
-#minUpper 0
-#minLower 0
-#minDigit 0
-#minPunct 0
+useCracklib 1
+minPoints 4
+minUpper 1
+minLower 1
+minDigit 1
+minPunct 1
 EOF
 mv README{,.check_pwd}
 popd
-- 
1.8.3.1

