From a6f919db5502fbb91a47b01000a297ab6c6c9a7c Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Tue, 5 Nov 2019 09:56:45 +0800
Subject: [PATCH 1/5] Update-package-versioning-for-TIS-format

---
 SPECS/initscripts.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index 4835491..3ac7b73 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -19,7 +19,7 @@ Requires:         gawk                       \
 Name:             initscripts
 Summary:          Basic support for legacy System V init scripts
 Version:          10.00.1
-Release:          1%{?dist}
+Release:          1.el8%{?_tis_dist}.%{tis_patch_ver}
 
 License:          GPLv2
 
-- 
1.8.3.1

