From 65f55952cedc1d514bbf03d85e529a17800dcc66 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Tue, 5 Nov 2019 10:21:42 +0800
Subject: [PATCH 5/5] added-ifup-alias-check-ipaddr

---
 SPECS/initscripts.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index a2696ca..d65f569 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -67,6 +67,7 @@ Patch10: 0010-ipv6-static-route-support.patch
 Patch11: 0011-ifup-eth-stop-waiting-if-link-is-up.patch
 Patch12: 0012-run-dhclient-as-daemon-for-ipv6.patch
 Patch13: 0013-ifup-alias-scope.patch
+Patch14: 0014-ifup-alias-check-ipaddr.patch
 
 # === PATCHES =================================================================
 
-- 
1.8.3.1

