From 885a175111d10b0892256cb3145d4c7313154ada Mon Sep 17 00:00:00 2001
From: chen-dq <chen.dq@neusoft.com>
Date: Fri, 15 Nov 2019 03:29:00 -0500
Subject: [PATCH 1/2] Update-package-versioning-for-TIS-format

---
 SPECS/python-eventlet.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python-eventlet.spec b/SPECS/python-eventlet.spec
index 6a19114..d195457 100644
--- a/SPECS/python-eventlet.spec
+++ b/SPECS/python-eventlet.spec
@@ -12,7 +12,7 @@
 
 Name:           python-%{pypi_name}
 Version:        0.18.4
-Release:        2%{?dist}
+Release:        2.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Highly concurrent networking library
 License:        MIT
 URL:            http://eventlet.net
-- 
2.18.1

