From 21db84dcb55f87c792a6d59cef0c68741a9d24b1 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:50:44 -0400
Subject: [PATCH 1/4] STX: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/sudo.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/sudo.spec b/SPECS/sudo.spec
index 2300d28..8a67560 100644
--- a/SPECS/sudo.spec
+++ b/SPECS/sudo.spec
@@ -1,7 +1,7 @@
 Summary: Allows restricted root access for specified users
 Name: sudo
 Version: 1.8.25p1
-Release: 4%{?dist}
+Release: 4.el8%{?_tis_dist}.%{tis_patch_ver}
 License: ISC
 Group: Applications/System
 URL: http://www.courtesan.com/sudo/
-- 
1.8.3.1

