From 11bc780fc8c7cfc7bbc59dcd84f3735c1cabfd30 Mon Sep 17 00:00:00 2001
From: Ludovic Beliveau <ludovic.beliveau@windriver.com>
Date: Mon, 30 May 2016 10:12:43 -0400
Subject: [PATCH 1/1] dhclient remove -1 arg

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 network-scripts/ifup-eth | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/network-scripts/ifup-eth b/network-scripts/ifup-eth
index 29f1161..0ac311c 100755
--- a/network-scripts/ifup-eth
+++ b/network-scripts/ifup-eth
@@ -215,7 +215,8 @@ if [ -n "${DYNCONFIG}" ] && [ -x /sbin/dhclient ]; then
     generate_lease_file_name
 
     # Initialize the dhclient args and obtain the hostname options if needed:
-    DHCLIENTARGS="${DHCLIENTARGS} ${ONESHOT} -q ${DHCLIENTCONF} -lf ${LEASEFILE} --restrict-interfaces -pf /run/dhclient-${DEVICE}.pid"
+    # DHCLIENTARGS="${DHCLIENTARGS} ${ONESHOT} -q ${DHCLIENTCONF} -lf ${LEASEFILE} --restrict-interfaces -pf /run/dhclient-${DEVICE}.pid"
+    DHCLIENTARGS="${DHCLIENTARGS} ${DHCLIENTCONF} -lf ${LEASEFILE} --restrict-interfaces -pf /run/dhclient-${DEVICE}.pid"
     set_hostname_options DHCLIENTARGS
 
     echo
-- 
1.8.3.1

