From 6ef93cd334969facf20502cf423524889abb61f9 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 4 Nov 2019 19:33:31 +0800
Subject: [PATCH 02/14] relocate-dhclient-leases-to-var-run

---
 network-scripts/network-functions | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/network-scripts/network-functions b/network-scripts/network-functions
index a2382e3..c4e1c63 100644
--- a/network-scripts/network-functions
+++ b/network-scripts/network-functions
@@ -65,11 +65,11 @@ get_uuid_by_config ()
 generate_lease_file_name ()
 {
     local ver=$1
-    LEASEFILE="/var/lib/dhclient/dhclient$ver-${DEVICE}.leases"
+    LEASEFILE="/var/run/dhclient$ver-${DEVICE}.leases"
     if [ -f $LEASEFILE ]; then
          return
     fi
-    LEASEFILE="/var/lib/dhclient/dhclient$ver-${UUID}-${DEVICE}.lease"
+    LEASEFILE="/var/run/dhclient$ver-${UUID}-${DEVICE}.lease"
 }
 
 generate_config_file_name ()
-- 
1.8.3.1

