From 6758af363f039ba4f7e6234d02feab008792a3cb Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 4 Nov 2019 20:04:22 +0800
Subject: [PATCH 06/14] force-delay-check-link-down

---
 network-scripts/network-functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/network-scripts/network-functions b/network-scripts/network-functions
index c4e1c63..db462bf 100644
--- a/network-scripts/network-functions
+++ b/network-scripts/network-functions
@@ -456,7 +456,7 @@ check_link_down ()
         ip link set dev $1 up >/dev/null 2>&1
     fi
     timeout=0
-    delay=10
+    delay=20
     [ -n "$LINKDELAY" ] && delay=$(($LINKDELAY * 2))
     while [ $timeout -le $delay ]; do
         [ "$(cat /sys/class/net/$REALDEVICE/carrier 2>/dev/null)" != "0" ] && return 1
-- 
1.8.3.1

