From 49a38fca4d1b710d54ecef879ce8e9b231522997 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 4 Nov 2019 20:10:56 +0800
Subject: [PATCH 08/14] sysconfig-affirmative-check-for-link-carrier

---
 network-scripts/network-functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/network-scripts/network-functions b/network-scripts/network-functions
index db462bf..0ab632d 100644
--- a/network-scripts/network-functions
+++ b/network-scripts/network-functions
@@ -459,7 +459,7 @@ check_link_down ()
     delay=20
     [ -n "$LINKDELAY" ] && delay=$(($LINKDELAY * 2))
     while [ $timeout -le $delay ]; do
-        [ "$(cat /sys/class/net/$REALDEVICE/carrier 2>/dev/null)" != "0" ] && return 1
+        [ "$(cat /sys/class/net/$REALDEVICE/carrier 2>/dev/null)" == "1" ] && return 1
         sleep 0.5
         timeout=$((timeout+1))
     done
-- 
1.8.3.1

