From 5cf9105829328f241376731d1630c089f5d5aa5e Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 4 Nov 2019 20:25:36 +0800
Subject: [PATCH 12/14] run-dhclient-as-daemon-for-ipv6

---
 network-scripts/ifup-eth | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/network-scripts/ifup-eth b/network-scripts/ifup-eth
index 5d25660..2084673 100755
--- a/network-scripts/ifup-eth
+++ b/network-scripts/ifup-eth
@@ -367,13 +367,14 @@ if is_true "${DHCPV6C}" && [ -x /sbin/dhclient ]; then
     echo -n $"Determining IPv6 information for ${DEVICE}..."
 
     # Initialize the dhclient args for IPv6 and obtain the hostname options if needed:
-    DHCLIENTARGS="-6 -1 ${DHCPV6C_OPTIONS} ${DHCLIENTCONF} -lf ${LEASEFILE} --restrict-interfaces -pf /run/dhclient6-${DEVICE}.pid ${DEVICE}"
+    DHCLIENTARGS="-6 ${DHCPV6C_OPTIONS} ${DHCLIENTCONF} -lf ${LEASEFILE} --restrict-interfaces -pf /run/dhclient6-${DEVICE}.pid ${DEVICE}"
     set_hostname_options DHCLIENTARGS
 
-    if /sbin/dhclient $DHCLIENTARGS; then
+    if /sbin/dhclient -1 $DHCLIENTARGS; then
         echo $" done."
     else
-        echo $" failed."
+        /sbin/dhclient $DHCLIENTARGS&
+        echo $" failed.  Retrying in background."
         if [ "${dhcpipv4}" = "good" -o -n "${IPADDR}" ]; then
             net_log "Unable to obtain IPv6 DHCP address ${DEVICE}." warning
         else
-- 
1.8.3.1

