From 59e30a344df4b661f30c0a5c629dbd13e9d88e8f Mon Sep 17 00:00:00 2001
From: Teresa Ho <teresa.ho@windriver.com>
Date: Mon, 17 Dec 2018 17:47:18 -0500
Subject: [PATCH 1/1] WRS: Patch13: ifup-alias-scope.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 network-scripts/ifup-aliases | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/network-scripts/ifup-aliases b/network-scripts/ifup-aliases
index 8a943c1..7735192 100755
--- a/network-scripts/ifup-aliases
+++ b/network-scripts/ifup-aliases
@@ -275,8 +275,14 @@ function new_interface ()
             fi
         fi
 
+        if [ "${parent_device}" = "lo" ]; then
+            SCOPE="scope host"
+        else
+            SCOPE=${SCOPE:-}
+        fi
+
         /sbin/ip addr add ${IPADDR}/${PREFIX} brd ${BROADCAST} \
-            dev ${parent_device} label ${DEVICE}
+            dev ${parent_device} ${SCOPE} label ${DEVICE}
 
         # update ARP cache of neighboring computers:
         if ! is_false "${ARPUPDATE}" && [ "${REALDEVICE}" != "lo" ]; then
-- 
1.8.3.1

