From e47e3faa2a3a35018e111dbd061a2e529cf77f9c Mon Sep 17 00:00:00 2001
From: Kevin Smith <kevin.smith@windriver.com>
Date: Tue, 17 Oct 2017 10:46:00 -0500
Subject: [PATCH 1/1] ipv6 static route support

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 network-scripts/ifup-routes | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/network-scripts/ifup-routes b/network-scripts/ifup-routes
index 517c2bb..1e2a345 100755
--- a/network-scripts/ifup-routes
+++ b/network-scripts/ifup-routes
@@ -32,9 +32,11 @@ handle_ip_file() {
     if [ "$type" != "$t" ]; then
         proto="-6"
     fi
+    # remove proto input from below so we can keep
+    # ipv6 routes in a route-<if> file as well.
     { cat "$file" ; echo ; } | while read line; do
         if [[ ! "$line" =~ $MATCH ]]; then
-            /sbin/ip $proto $type add $line
+            /sbin/ip $type add $line
         fi
     done
 }
-- 
1.8.3.1

