From 76a5f892c132eed05a6cbffbdba3306e50b6a672 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Thu, 5 Oct 2017 12:40:38 -0400
Subject: [PATCH 7/9] WRS: Patch10-run-ifdown-on-all-interfaces.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 etc/rc.d/init.d/network | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/etc/rc.d/init.d/network b/etc/rc.d/init.d/network
index 798e0d0..0bf5aa6 100755
--- a/etc/rc.d/init.d/network
+++ b/etc/rc.d/init.d/network
@@ -232,6 +232,9 @@ stop)
         if ! check_device_down $DEVICE; then
             action $"Shutting down interface $i: " ./ifdown $i boot
             [ $? -ne 0 ] && rc=1
+        else
+            action $"Shutting down non-UP interface $i: " ./ifdown $i boot
+            logger $"Running ifdown on non-UP interface $i"
         fi
         )
     done
-- 
1.8.3.1

