From 2233043da5631c7d205e781e467b3e03a53d5b25 Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Sat, 2 Nov 2019 14:22:47 +0800
Subject: [PATCH 1/2] spec-include-TiS-patches

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/dhcp.spec | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 7bbd360..f0084b1 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -80,6 +80,11 @@ Patch43:  dhcp-4.3.6-reference_count_overflow.patch
 Patch44:  dhcp-replay_file_limit.patch
 Patch45:  dhcp-4.2.5-expiry_before_renewal_v2.patch
 
+# WRS
+Patch101: dhclient-restrict-interfaces-to-command-line.patch
+Patch102: dhclient-ipv6-bind-to-interface.patch
+Patch103: dhclient-ipv6-conditionally-set-hostname.patch
+
 BuildRequires: autoconf
 BuildRequires: automake
 BuildRequires: libtool
@@ -344,6 +349,11 @@ rm bind/bind.tar.gz
 # https://bugzilla.redhat.com/show_bug.cgi?id=1647786
 %patch45 -p1 -b .t2-expirity
 
+# WRS
+%patch101 -p1
+%patch102 -p1
+%patch103 -p1
+
 # DHCLIENT_DEFAULT_PREFIX_LEN  64 -> 128
 # https://bugzilla.gnome.org/show_bug.cgi?id=656610
 sed -i -e 's|DHCLIENT_DEFAULT_PREFIX_LEN 64|DHCLIENT_DEFAULT_PREFIX_LEN 128|g' includes/site.h
-- 
2.7.4

