From 91c0943613fa00e7fd446759899fa6ad534f390f Mon Sep 17 00:00:00 2001
Message-Id: <91c0943613fa00e7fd446759899fa6ad534f390f.1536700525.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Tue, 15 Aug 2017 14:04:06 -0400
Subject: [PATCH 1/1] Update package versioning for TIS format

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/libvirt-python.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libvirt-python.spec b/SPECS/libvirt-python.spec
index 571b4a8..99afd2e 100644
--- a/SPECS/libvirt-python.spec
+++ b/SPECS/libvirt-python.spec
@@ -34,7 +34,7 @@
 Summary: The libvirt virtualization API python2 binding
 Name: libvirt-python
 Version: 4.5.0
-Release: 1%{?dist}%{?extra_release}
+Release: 1%{?_tis_dist}.%{tis_patch_ver}
 Source0: http://libvirt.org/sources/python/%{name}-%{version}.tar.gz
 Url: http://libvirt.org
 License: LGPLv2+
-- 
2.18.1

