From 50754742b9b7886cf28920d4f3dc88af652fa133 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Wed, 6 Nov 2019 15:45:19 +0800
Subject: [PATCH 2/2] Build-python-wheel

---
 SPECS/libvirt-python.spec | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/SPECS/libvirt-python.spec b/SPECS/libvirt-python.spec
index 99afd2e..fd4eb0b 100644
--- a/SPECS/libvirt-python.spec
+++ b/SPECS/libvirt-python.spec
@@ -36,6 +36,7 @@ Name: libvirt-python
 Version: 4.5.0
 Release: 1%{?_tis_dist}.%{tis_patch_ver}
 Source0: http://libvirt.org/sources/python/%{name}-%{version}.tar.gz
+Patch0: 0001-Fix-setup-import-to-support-building-wheel.patch
 Url: http://libvirt.org
 License: LGPLv2+
 Group: Development/Libraries
@@ -46,10 +47,14 @@ BuildRequires: libvirt-devel >= 4.5.0-1%{?dist}
 BuildRequires: python2-devel
 BuildRequires: python2-nose
 BuildRequires: python2-lxml
+BuildRequires: python2-pip
+BuildRequires: python2-wheel
 %else
 BuildRequires: python-devel
 BuildRequires: python-nose
 BuildRequires: python-lxml
+BuildRequires: python2-pip
+BuildRequires: python2-wheel
 %endif
 %endif
 %if %{with_python3}
@@ -109,6 +114,7 @@ of recent versions of Linux (and other OSes).
 
 %prep
 %setup -q
+%patch0 -p1
 
 # Unset execute bit for example scripts; it can introduce spurious
 # RPM dependencies, like /usr/bin/python which can pull in python2
@@ -123,6 +129,7 @@ exit 1
 
 %if %{with_python2}
 CFLAGS="$RPM_OPT_FLAGS" %{__python} setup.py build
+CFLAGS="$RPM_OPT_FLAGS" %{__python2} setup.py bdist_wheel
 %endif
 %if %{with_python3}
 CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py build
@@ -131,6 +138,8 @@ CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py build
 %install
 %if %{with_python2}
 %{__python} setup.py install --skip-build --root=%{buildroot}
+mkdir -p $RPM_BUILD_ROOT/wheels
+install -m 644 dist/*.whl $RPM_BUILD_ROOT/wheels/
 %endif
 %if %{with_python3}
 %{__python3} setup.py install --skip-build --root=%{buildroot}
@@ -153,6 +162,16 @@ CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py build
 %{python_sitearch}/libvirt_lxc.py*
 %{python_sitearch}/libvirtmod*
 %{python_sitearch}/*egg-info
+
+%package wheels
+Summary: %{name} wheels
+
+%description wheels
+Contains python wheels for %{name}
+
+%files wheels
+/wheels/*
+
 %endif
 
 %if %{with_python3}
-- 
1.8.3.1

