From 8feb14092d9db4aacd907bd8f76d90931f537eda Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Wed, 17 Oct 2018 15:25:33 -0400
Subject: [PATCH 2/2] Build python wheel

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/libvirt-python.spec | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/SPECS/libvirt-python.spec b/SPECS/libvirt-python.spec
index 99afd2e..0e37d46 100644
--- a/SPECS/libvirt-python.spec
+++ b/SPECS/libvirt-python.spec
@@ -36,6 +36,7 @@ Name: libvirt-python
 Version: 4.5.0
 Release: 1%{?_tis_dist}.%{tis_patch_ver}
 Source0: http://libvirt.org/sources/python/%{name}-%{version}.tar.gz
+Patch0: 0001-Fix-setup-import-to-support-building-wheel.patch
 Url: http://libvirt.org
 License: LGPLv2+
 Group: Development/Libraries
@@ -56,6 +57,8 @@ BuildRequires: python-lxml
 BuildRequires: python3-devel
 BuildRequires: python3-nose
 BuildRequires: python3-lxml
+BuildRequires: python3-pip
+BuildRequires: python3-wheel
 %endif
 
 # Don't want provides for python shared objects
@@ -109,6 +112,7 @@ of recent versions of Linux (and other OSes).
 
 %prep
 %setup -q
+%patch0 -p1
 
 # Unset execute bit for example scripts; it can introduce spurious
 # RPM dependencies, like /usr/bin/python which can pull in python2
@@ -126,6 +130,7 @@ CFLAGS="$RPM_OPT_FLAGS" %{__python} setup.py build
 %endif
 %if %{with_python3}
 CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py build
+CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py bdist_wheel
 %endif
 
 %install
@@ -134,6 +139,8 @@ CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py build
 %endif
 %if %{with_python3}
 %{__python3} setup.py install --skip-build --root=%{buildroot}
+mkdir -p $RPM_BUILD_ROOT/wheels
+install -m 644 dist/*.whl $RPM_BUILD_ROOT/wheels/
 %endif
 
 %check
@@ -169,6 +176,16 @@ CFLAGS="$RPM_OPT_FLAGS" %{__python3} setup.py build
 %{python3_sitearch}/__pycache__/libvirtaio.cpython-*.py*
 %{python3_sitearch}/libvirtmod*
 %{python3_sitearch}/*egg-info
+
+%package wheels
+Summary: %{name} wheels
+
+%description wheels
+Contains python wheels for %{name}
+
+%files wheels
+/wheels/*
+
 %endif
 
 %changelog
-- 
2.18.1

