From b6a87543c2813a89fec2ffa91757082286a33e93 Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Sat, 2 Nov 2019 15:51:51 +0800
Subject: [PATCH] dhclient-dhcp6-set-hostname

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SOURCES/dhclient-script | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/SOURCES/dhclient-script b/SOURCES/dhclient-script
index 5f58112..8d1adf2 100644
--- a/SOURCES/dhclient-script
+++ b/SOURCES/dhclient-script
@@ -647,6 +647,10 @@ dh6config() {
             add_ipv6_addr_with_DAD
 
             make_resolv_conf
+
+            if [ -n "${new_fqdn_hostname}" ] && need_hostname; then
+                hostname ${new_fqdn_hostname} || echo "See -nc option in dhclient(8) man page."
+            fi
             ;;
 
         RENEW6|REBIND6)
@@ -663,6 +667,10 @@ dh6config() {
                [ ! "${new_dhcp6_domain_search}" = "${old_dhcp6_domain_search}" ]; then
                 make_resolv_conf
             fi
+
+            if [ -n "${new_fqdn_hostname}" ] && need_hostname; then
+                hostname ${new_fqdn_hostname} || echo "See -nc option in dhclient(8) man page."
+            fi
             ;;
 
         DEPREF6)
-- 
2.7.4

