From 667cc2ccdd4451ce9d943064d714bcf1a8d1e4ed Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 26 Sep 2016 17:40:54 -0400
Subject: [PATCH 09/15] Update package versioning for TIS format

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/setup.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index acd6d95..2b16ffb 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -1,7 +1,7 @@
 Summary: A set of system configuration and setup files
 Name: setup
 Version: 2.12.2
-Release: 2%{?dist}
+Release: 2.el8%{?_tis_dist}.%{tis_patch_ver}
 License: Public Domain
 Group: System Environment/Base
 URL: https://pagure.io/setup/
-- 
1.8.3.1

