From e3aba238a88f73392d832b5ae366466f783adce4 Mon Sep 17 00:00:00 2001
From: zhipengl <zhipengs.liu@intel.com>
Date: Sun, 18 Nov 2018 10:06:23 +0000
Subject: [PATCH] WRS: 0001-patch-passwd-group.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/setup.spec | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 2b16ffb..c3428e5 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -13,12 +13,17 @@ Requires: system-release
 Conflicts: filesystem < 3
 Conflicts: initscripts < 4.26, bash <= 2.0.4-21
 
+Patch1001: 0001-Change-group-passwd-and-uidgid.patch
+
 %description
 The setup package contains a set of important system configuration and
 setup files, such as passwd, group, and profile.
 
 %prep
 %setup -q
+
+%patch1001 -p1
+
 ./shadowconvert.sh
 
 %build
-- 
1.8.3.1

