From 63375de22771a9384eb9c4a90abc8ee2143a7721 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:30:57 -0400
Subject: [PATCH] WRS: 0001-Update-package-versioning-for-TIS-format.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/tpm2-tools.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/tpm2-tools.spec b/SPECS/tpm2-tools.spec
index 8673838..c0a9d64 100644
--- a/SPECS/tpm2-tools.spec
+++ b/SPECS/tpm2-tools.spec
@@ -1,6 +1,6 @@
 Name: tpm2-tools
 Version: 3.1.1
-Release: 4%{?dist}
+Release: 4.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary: A TPM2.0 testing tool build upon TPM2.0-TSS
 
 License: BSD
-- 
1.8.3.1

