From e54d19f53dcd78d0aaceb30d1fb89eebefc1c066 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:30:56 -0400
Subject: [PATCH 1/2] WRS: disable-socket-mode-TCTI.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/tpm2-tools.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/tpm2-tools.spec b/SPECS/tpm2-tools.spec
index c0a9d64..f600d3a 100644
--- a/SPECS/tpm2-tools.spec
+++ b/SPECS/tpm2-tools.spec
@@ -35,7 +35,7 @@ tpm2-tools is a batch of testing tools for tpm2.0. It is based on tpm2-tss.
 %autosetup -p1 -n %{name}-%{version}
 
 %build
-%configure --prefix=/usr --disable-static --disable-silent-rules
+%configure --prefix=/usr --disable-static --disable-silent-rules --without-tcti-socket
 %make_build
 
 %install
-- 
1.8.3.1

