From f7ac0c586ee46b67c7b5a541ee823f459e19c5c6 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Mon, 27 Nov 2017 08:35:10 -0500
Subject: [PATCH 1/1] WRS: 8000-TiS-tboot.patch

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/tboot.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/tboot.spec b/SPECS/tboot.spec
index 4a2db5b..c412cf7 100644
--- a/SPECS/tboot.spec
+++ b/SPECS/tboot.spec
@@ -1,7 +1,7 @@
 Summary:        Performs a verified launch using Intel TXT
 Name:           tboot
 Version:        1.9.7
-Release:        2%{?dist}
+Release:        2.el8%{?_tis_dist}.%{tis_patch_ver}
 Epoch:          1
 
 Group:          System Environment/Base
-- 
1.8.3.1

