From 16a82ea84332a117c4524caaa4209b912e18e888 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Wed, 6 Dec 2017 08:47:12 -0500
Subject: [PATCH 1/1] TiS tboot

Signed-off-by: chendongqi <chen.dq@neusoft.com>
---
 SPECS/tboot.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/tboot.spec b/SPECS/tboot.spec
index c412cf7..9e09659 100644
--- a/SPECS/tboot.spec
+++ b/SPECS/tboot.spec
@@ -9,6 +9,8 @@ License:        BSD
 URL:            http://sourceforge.net/projects/tboot/
 Source0:        http://downloads.sourceforge.net/%{name}/%{name}-%{version}.tar.gz
 
+Patch999: ../patches/1000-tboot-for-tis.patch
+
 BuildRequires:  trousers-devel
 BuildRequires:  openssl-devel
 ExclusiveArch:  %{ix86} x86_64
@@ -21,6 +23,8 @@ and verified launch of an OS kernel/VMM.
 %prep
 %setup -q
 
+%patch999 -p1
+
 # do not override OPTFLAGS
 sed -i -e 's/-march=i686//' Config.mk
 
-- 
1.8.3.1

