From 9ffe510612701aa6bb2f72bfefda53c3ce886bc3 Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Fri, 1 Nov 2019 13:36:33 +0800
Subject: [PATCH 2/3] TiS-tboot

---
 SPECS/tboot.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/tboot.spec b/SPECS/tboot.spec
index c412cf7..9e09659 100644
--- a/SPECS/tboot.spec
+++ b/SPECS/tboot.spec
@@ -9,6 +9,8 @@ License:        BSD
 URL:            http://sourceforge.net/projects/tboot/
 Source0:        http://downloads.sourceforge.net/%{name}/%{name}-%{version}.tar.gz
 
+Patch999: ../patches/1000-tboot-for-tis.patch
+
 BuildRequires:  trousers-devel
 BuildRequires:  openssl-devel
 ExclusiveArch:  %{ix86} x86_64
@@ -21,6 +23,8 @@ and verified launch of an OS kernel/VMM.
 %prep
 %setup -q
 
+%patch999 -p1
+
 # do not override OPTFLAGS
 sed -i -e 's/-march=i686//' Config.mk
 
-- 
1.8.3.1

