From 77f000fa64b180d3ddba2598a13fca9b0c212d0c Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Tue, 5 Nov 2019 19:59:46 +0800
Subject: [PATCH 2/2] 0001-Update-package-versioning-for-TIS-format

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/bash.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/bash.spec b/SPECS/bash.spec
index 52a5d03..4f2b7ab 100644
--- a/SPECS/bash.spec
+++ b/SPECS/bash.spec
@@ -7,7 +7,7 @@
 Version: %{baseversion}%{patchleveltag}
 Name: bash
 Summary: The GNU Bourne Again shell
-Release: 7%{?dist}
+Release: 7.el8%{?_tis_dist}.%{tis_patch_ver}
 License: GPLv3+
 Url: https://www.gnu.org/software/bash
 Source0: https://ftp.gnu.org/gnu/bash/bash-%{baseversion}.tar.gz
-- 
2.7.4

