From aae4b710faadd16de4e2255e7b4cbc4868e62422 Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Fri, 1 Nov 2019 17:32:08 +0800
Subject: [PATCH 2/2] spec-TiS-bash-history

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/bash.spec | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/SPECS/bash.spec b/SPECS/bash.spec
index 100e440..425672b 100644
--- a/SPECS/bash.spec
+++ b/SPECS/bash.spec
@@ -95,6 +95,10 @@ Patch131: bash-4.4-case-in-command-subst.patch
 
 Patch132: bash-4.4-coverity.patch
 
+# Patches from WindRiver
+Patch500: bash-history-syslog.patch
+Patch501: bash-history-exit-child-on-parent-death.patch
+
 BuildRequires: texinfo bison
 BuildRequires: ncurses-devel
 BuildRequires: autoconf, gettext
@@ -126,6 +130,10 @@ This package contains documentation files for %{name}.
 %prep
 %autosetup -n %{name}-%{baseversion} -p1
 
+# WindRiver patches
+%patch500 -p1 -b .history-syslog
+%patch501 -p1 -b .history-exit-child-on-parent-death
+
 echo %{version} > _distribution
 echo %{release} > _patchlevel
 
-- 
2.7.4

