From 374cd73df6711cae546189ebd9a023a695e2a829 Mon Sep 17 00:00:00 2001
From: lilong-neu <lilong-neu@neusoft.com>
Date: Tue, 5 Nov 2019 19:58:19 +0800
Subject: [PATCH 1/2] spec-TiS-bash-history

Signed-off-by: lilong-neu <lilong-neu@neusoft.com>
---
 SPECS/bash.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/bash.spec b/SPECS/bash.spec
index 7e3549a..52a5d03 100644
--- a/SPECS/bash.spec
+++ b/SPECS/bash.spec
@@ -95,6 +95,10 @@ Patch131: bash-4.4-case-in-command-subst.patch
 
 Patch132: bash-4.4-coverity.patch
 
+# Patches from WindRiver
+Patch500: bash-history-syslog.patch
+Patch501: bash-history-exit-child-on-parent-death.patch
+
 BuildRequires: texinfo bison
 BuildRequires: ncurses-devel
 BuildRequires: autoconf, gettext
-- 
2.7.4

